### PR TITLE
FND-912: New implementation of reset action

### PR DESCRIPTION
**Pull request by NicHaley**
**What**
Adds a different implementation for the `reset` action. This was done to fix some bugs with the old implementation, as well as add more features to the `reset` action. Now, the `reset` action can also optionally accept an object where `metaFields`, `greeting`, and `language` can be modified for new users. It can also accept a new field `resetChatHistory` which can be set to `false` if the client wishes to reset the window without creating a new `chatter`.

**How**
Previously, when the `reset` action was triggered on Embed, a postMessage event was sent to Chat. This would tell Chat to refresh the Redux state and restart, but was buggy for a number of reasons.

In the new implementation, the Chat iFrame will be removed on reset, and replaced with a new Chat window with query parameter `reset=1`. Chat will recognize that it is in "reset" mode, and create a new session on load.

**Testing**
To test, `nh-reset-fix` should be checked out in Chat.

This should be tested in a few different scenarios:
- Try resetting on page load. Previously, this would case a race condition in the old implementation (this should no longer happen).
- Reset with and without new data
- Reset after Chat has already been opened

---
- [x] PR title follows the format \<Jira Card\>: \<Change description\>
- [x] PR description present and accurate
- [x] PR is linked to a Jira card has `impact` field filled in
- [x] Unit tests cover code changes
- [ ] Changes have been tested on staging `embed-testing.svc.ada.support` site
- [ ] Changes have gone through design review if required
- [ ] Deployment plan and/or bridge code description is included in PR if required
- [x] Changes are CATO/WCAG compliant and support IE9+
- [x] A [Release Note](https://www.notion.so/adasupport/Creating-Release-Notes-36906dfa8a2b4f10a31dc23b8d46681a) has been drafted and published after merging to master

### CODE REVIEW

**Comment by NicHaley** [comment]
TODO: Initial props set by client are now being set on init to state. We need to do this so that these values can be easily changed on reset. _However_, we now have 2 sources of truth: the values in props and the values in state. Developers should just rely on state, however this might not be clear. We might want to consider renaming the input props so that they explicitly indicate they should not be used. Ex. `handleInitProp` instead of `handle`.

**Comment by NicHaley** [comment]
> TODO: Initial props set by client are now being set on init to state. We need to do this so that these values can be easily changed on reset. _However_, we now have 2 sources of truth: the values in props and the values in state. Developers should just rely on state, however this might not be clear. We might want to consider renaming the input props so that they explicitly indicate they should not be used. Ex. `handleInitProp` instead of `handle`.

Done

**Comment by NicHaley** [comment]
reviewme

**Comment by NicHaley** [comment]
reviewme

File: src/index.ts [hunk 1]
```diff
@@ -34,6 +34,66 @@ declare global {
   }
 }
 
+/**
+ * Intial inputs passed in by client during instantiation
+ */
+export interface IntracefaceStartOptions {
```
Comment (lozog) [inline src/index.ts]: typo?

File: src/index.ts [hunk 2]
```diff
@@ -86,7 +146,7 @@ const adaEmbed = Object.freeze({
   /**
    * Setup Ada Embed
    */
-  [ADA_EVENT_START]: (options: any) => {
+  [ADA_EVENT_START]: (options: IntracefaceStartOptions) => {
```
Comment (lozog) [inline src/index.ts]: typo

File: src/components/AppWrapper/index.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import {
+  DEFAULT_BUTTON_POSITION
+} from "constants/configuration";
+import App from "./App";
+import { Component, h } from "preact";
+import Client from "models/Client";
+import { IntracefaceStartOptions as InterfaceAppWrapper } from "src/index";
```
Comment (lozog) [inline src/components/AppWrapper/index.tsx]: typo

File: src/index.ts (hunk 1 above)
Comment (NicHaley) [reply src/index.ts]: aha good catch!

**Pull request merged**